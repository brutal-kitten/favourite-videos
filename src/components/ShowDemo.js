import React, { Component } from 'react';

class ShowDemo extends Component {

  handleClick = (event) => {
    event.preventDefault();
    this.props.showDemo();
  }

  render() {

    return (
      <div className="showDemo" >
        <button type="button" onClick={(event) => this.handleClick(event)}>Show demolist</button>
      </div>
    )
  }
}

export default ShowDemo;
