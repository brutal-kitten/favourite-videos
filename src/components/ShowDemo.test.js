import React from 'react';
import ReactDOM from 'react-dom';
import ShowDemo from './ShowDemo.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('should display a button', () => {
  const tree = TestRenderer.create(
    <ShowDemo />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe ('shoud call handleClick function after click', () => {
  const mockShowDemo = jest.fn();
  const props = {
    showDemo: mockShowDemo
};

  const wrapper = shallow(<ShowDemo {...props}/>);

  wrapper.find('button').simulate('click', {
    preventDefault: () => {}
  });

  it('should call showDemo function', () => {
    expect(mockShowDemo.mock.calls.length).toBe(1);
  });
})
