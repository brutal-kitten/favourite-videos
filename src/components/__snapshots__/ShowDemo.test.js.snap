// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a button 1`] = `
<div
  className="showDemo"
>
  <button
    className="buttonShowDemo"
    onClick={[Function]}
    tabIndex="0"
    type="button"
  >
    Show demolist
  </button>
</div>
`;
